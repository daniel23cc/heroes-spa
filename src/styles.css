/* Estilos del componente del menú */
.menu {
    font-family: 'Roboto', sans-serif;
    background-image: url('../../fondos/fondo.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center center;
}

.logo {
    margin-right: 10px;
}

/* Estilos del componente de la página de login */
.login-page-container {
    height: 100vh;
    background-image: url('../../fondos/loginfondo.jpg');
    background-size: cover;
    background-position: center;
}

.login-page-container {
    font-family: 'Sriracha-Regular', sans-serif;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    text-align: right;
    padding-right: 15%;
}

.login-page-container h1 {
    animation: slideInRight 1s ease;
    font-weight: bold;
    font-family: 'Sriracha-Regular';
}

.login-page-container button {
    animation: slideInRight 1s ease;
    font-weight: bold;
}

.my-nav {
    animation-duration: 0.5s;
    /* Cambia el valor por el deseado (en segundos) */
}


@keyframes slideInRight {
    from {
        transform: translateX(100%);
    }

    to {
        transform: translateX(0);
    }
}


@font-face {
    font-family: 'Sriracha-Regular';
    src: url('../../fonts/Sriracha-Regular.woff2') format('woff2'),
         url('../../fonts/Sriracha-Regular.wof') format('woff');
    font-weight: 400;
    font-style: normal;
  }