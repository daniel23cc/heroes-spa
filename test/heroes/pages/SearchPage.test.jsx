import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes"

const mockedUseNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate
}))

describe('Pruebas en SearchPage', () => {
    beforeEach(() => jest.clearAllMocks())

    test('debe de mosrarse con valores defecto', () => {
        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar a batman y el input con el valor de queryString', () => {
        const { container } = render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        )
        //screen.debug()
        //expect(container).toMatchSnapshot()

        const input = screen.getByRole('textbox')
        expect(input.value).toBe('batman')

        const img = screen.getByRole('img')
        expect(img.src).toContain('/heroes/dc-batman.jpg')

        const alertDanger = screen.getByLabelText('alert-danger')
        expect(alertDanger.style.display).toBe('none')
        //screen.debug()

    })

    test('Debe de mostrar error si no encuentra hero (batman123)', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        )

        //screen.debug()

        const alertDanger = screen.getByLabelText('alert-danger')
        expect(alertDanger.style).not.toBe('none')

    })

    test('navigate a pantalla nueva', () => {
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        )
        screen.debug()
        const input = screen.getByRole('textbox')
        fireEvent.change(input, { target: { name: 'searchText', value: 'superman' } })

        const input2 = screen.getByRole('form')
        fireEvent.submit(input2)

        expect(mockedUseNavigate).toHaveBeenCalledWith('?q=superman')

    })


})