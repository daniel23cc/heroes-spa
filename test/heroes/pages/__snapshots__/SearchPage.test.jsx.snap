// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchPage debe de mosrarse con valores defecto 1`] = `
<div>
  <h1>
    Buscar
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        BÃºsqueda
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Buscar un heroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Buscar un heroe
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__bounce"
        style="display: none;"
      >
        No existe el heroe 
        <b />
      </div>
    </div>
  </div>
</div>
`;
